
#region DEPRECATED

public static void Operar()
        {
            string dato;
            int numero;

            try
            {
                dato = Console.ReadLine();
                numero = int.Parse(dato);
                numero = numero / numero;
            }
            catch (Exception ex)
            {
                //Dado en la clase: Operaciones para guardar informacion de la excepcion
                

                throw new Exception("Se quedo sin internet"); //Fuerza la excepcion
            }
        }

        public static void LanzarError() {
            throw new Exception("Se quedò sin memoria RAM"); //El throw no necesita try
        }
        //Dividir sistema que uno muestre mensaje y otra muestre error, o todo en uno

        static void Main(string[] args)
        {
            //string dato;
            //int numero;

            try
            {
                //Solamente se pone lo que puede dar error
                /*Pegado en el metodo static abajo
                 * dato = Console.ReadLine();
                numero = int.Parse(dato);
                numero = numero / numero;
                 * */
                LanzarError(); // Muestra de una que no hay ram, se va del try luego de ejecutarse
                Operar();
            }

            catch (FormatException ex)
            {
                Console.Write("Error de formato: " + ex.Message);
            }
            catch (DivideByZeroException ex)
            {
                Console.Write("No se puede dividir por 0: " + ex.Message);
            }
            catch (Exception ex) //nombre de la excepcion ex
            {
                //si ocurre un error, viene al catch
                //siempre devuelve un objeto de tipo exception

                Console.Write("General: " + ex.Message);

                //throw; // << sacar
            }
            finally{ // < Luego de todos los catch, siempre se ejecuta el finally
                //Se usa para liberar recursos o mostrar un mensaje final

            }
            //LanzarError(); // Va al error donde cayo en la linea
            Console.ReadKey();

        }

#endregion 